astropy/io/fits/diff.py

```diff
@@ -262,6 +276,9 @@ def __init__(self, a, b, ignore_keywords=[], ignore_comments=[],
         ignore_blank_cards : bool, optional
             Ignore all cards that are blank, i.e. they only contain
             whitespace (default: True).
+
+        tolerance : float, optional
```

Comment:
The deprecation of `tolerance` should now follow the same implementation similar to #5171, utilizing a new decorator added in #5214.